lib/api/model/model.dart

```diff
@@ -120,6 +78,41 @@ class User {
   Map<String, dynamic> toJson() => _$UserToJson(this);
 }
 
+/// As in [User.profileData].
+@JsonSerializable(fieldRename: FieldRename.snake)
+class ProfileFieldUserData {
+  final String value;
+  final String? renderedValue;
+
+  ProfileFieldUserData({required this.value, this.renderedValue});
+
+  factory ProfileFieldUserData.fromJson(Map<String, dynamic> json) =>
+      _$ProfileFieldUserDataFromJson(json);
+
+  Map<String, dynamic> toJson() => _$ProfileFieldUserDataToJson(this);
+}
+
+/// As in [User.role].
+@JsonEnum(fieldRename: FieldRename.snake, valueField: "apiValue")
```

Comment:
You're right, I got too carried away reusing existing boilerplate. It feels like every json annotation we have does use `FieldRename.snake` but on inspection it looks like `EmojiSet` uses `FieldRename.kebab`, and now this one.

Reply:
Yeah. We almost always use `FieldRename.snake` because the API is very nearly 100% consistent about naming things in snake case; for that reason I'm generally in favor of including it even if the existing set of names are all single-word, just so that the right thing will naturally happen if a new multi-word name is later added.

But the "Google blob" emojiset is called `google-blob`, so that enum is kebab case. And then there's a bunch of enums like this where the API uses numeric values instead of named values, and I think this PR introduces our first two of those in zulip-flutter. So for those we'll be using `valueField`, so `fieldRename` won't do anything even in principle (and not only through the happenstance of having only single-word names so far).